lib/Backend/FlowGraph.cpp

```diff
@@ -4431,6 +4431,311 @@ BasicBlock::CleanUpValueMaps()
+        {
+            return lastBranchTarget != nullptr;
+        }
+        if (instr->GetDst())
+        {
+            if (!instr->GetDst()->IsRegOpnd()) // complex dstOpnd, stop.
+            {
+                return lastBranchTarget != nullptr;
+            }
+            IR::RegOpnd *dst = instr->GetDst()->AsRegOpnd();
+            if (currentLabel)
+            {
+                BasicBlock * currentBlock = currentLabel->GetBasicBlock();
+                // If the dstOpnd is defined and used in the same block, it is legal, but if the successor blocks use it, stop.
+                if (currentBlock->succUses && currentBlock->succUses->Test(dst->GetStackSym()->m_id))
+                {
+                    return lastBranchTarget != nullptr;
+                }
+            }
+            if (instr->GetSrc1())
+            {
+                // If the src of the load instruction is produced in the currentBlock, its value will not be available in the valuetable, check in local valuetable
+                Value **localValue = localSymToValueMap->FindOrInsertNew(instr->GetDst()->GetSym());
+                if (instr->GetSrc1()->GetSym())
+                {
+                    Value **localSrc1Value = localSymToValueMap->Get(instr->GetSrc1()->GetSym());
+                    *localValue = localSrc1Value == nullptr ? this->globOptData.FindValue(instr->GetSrc1()->GetSym()) : *localSrc1Value;
+                }
+                else
+                {
+                    ValueType src1Value = instr->GetSrc1()->GetValueType();
+                    if (src1Value.IsUndefined())
+                    {
+                        *localValue = globOpt->NewGenericValue(ValueType::Undefined, instr->GetDst());
+                    }
+                    // MGTODO : handle other constants
+                    else
+                    {
+                        return lastBranchTarget != nullptr;
+                    }
+                }
+            }
+        }
+        if (instr->IsLabelInstr())
+        {
+            if (instr->AsLabelInstr()->m_isLoopTop)
+            {
+                // don't cross over to loops
+                return lastBranchTarget != nullptr;
+            }
+            currentLabel = instr->AsLabelInstr();
+        }
+        if (instr->IsBranchInstr())
+        {
+            IR::BranchInstr* branch = instr->AsBranchInstr();
+            IR::LabelInstr* branchTarget = nullptr;
+            if (branch->IsUnconditional())
+            {
+                branchTarget = branch->GetTarget();
+                if (!branchTarget)
+                {
+                    return lastBranchTarget != nullptr;
+                }
+                if (branchTarget && branchTarget->m_isLoopTop)
+                {
+                    return lastBranchTarget != nullptr;
+                }
+            }
+            else
+            {
+                if (branch->GetTarget()->m_isLoopTop)
```

Comment:
>branch->GetTarget()->m_isLoopTop [](start = 20, length = 32)

nit: could move this check before checking the conditionality of the branch

Reply:
GetTarget() can be nullptr

Reply:
Oh ya, I can move inspite of that.